nowcasting_dataset/data_sources/gsp/gsp_data_source.py

```diff
@@ -124,35 +124,52 @@ def get_locations(self, t0_datetimes: pd.DatetimeIndex) -> Tuple[List[Number], L
         Returns: list of x and y locations
 
         """
-        # Pick a random GSP for each t0_datetime, and then grab
-        # their geographical location.
-        x_locations = []
-        y_locations = []
+        total_gsp_nan_count = self.gsp_power.isna().sum().sum()
+        if total_gsp_nan_count == 0:
 
-        # TODO: Issue 305: Speed up this function by removing this for loop?
-        for t0_dt in t0_datetimes:
+            # get random gps metadata
+            indexes = list(np.random.choice(range(len(self.metadata)), size=len(t0_datetimes)))
```

Comment:
Should we use `self.rng` instead of `np.random`?  Also, it may be a tiny bit faster to use `integers()` instead of `choice()`.  Something like this:

```suggestion
            indexes = list(self.rng.integers(low=0, high=len(self.metadata), size=len(t0_datetimes)))
```

Reply:
Probably have to check that doesnt give repated values. 

However this is in the code that most of the time is used now, so I'm slightly tempted to leave it

Reply:
ah ok, yea sorry got confused thinking about GSPs for one example, but yea youa re right - 'rng.integers' will work